doit/doit_cmd.py

```diff
@@ -52,18 +52,28 @@ class DoitMain(object):
                  Strace, TabCompletion, ResetDep)
 
     def __init__(self, task_loader=None,
-                 config_filenames='doit.cfg',
+                 config_filenames=['pyproject.toml', 'doit.cfg'],
                  extra_config=None):
         self.task_loader = task_loader
 
-        # combine config option from INI files and API
+        # combine config option from INI/TOML files and API
         self.config = defaultdict(dict)
         if extra_config:
             for section, items in extra_config.items():
                 self.config[section].update(items)
-        ini_config = self.load_config_ini(config_filenames)
-        for section in ini_config.sections():
-            self.config[section].update(ini_config[section].items())
+
+        if isinstance(config_filenames, str):
+            config_filenames = [config_filenames]
+
+        for config_filename in config_filenames:
+            if str(config_filename).lower().endswith('.toml'):
```

Comment:
casting to a string, here, as it can be a `Path` (python 3.6.1+) or `bytes` (python 3.7).